### ISSUE TITLE
[MASSEMBLY-594] Add ability to suppress "already added, skipping" messages

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#756]
**[monkegoist](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=monkegoist)** opened **[MASSEMBLY-594](https://issues.apache.org/jira/browse/MASSEMBLY-594?redirect=false)** and commented

A common multi-module project structure would define most of the project dependencies in the parent POM thus allowing all the child modules to inherit them. When packaging all the project dependencies using \<moduleSet>, you'd see a lot of messages like this (and in case of big project the number is huge):

```
[INFO] lib/log4j-1.2.15.jar already added, skipping
[INFO] lib/mail-1.4.jar already added, skipping
[INFO] lib/jms-1.1.jar already added, skipping
```

It would be great to either have an option to suppress such info messages, or to have them moved to DEBUG log level.

See example of configuration here: http://stackoverflow.com/questions/4912738/maven-assembly-include-all-sub-module-dependencies-without-already-added-skipp

---

5 votes, 1 watchers

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

FIrst based on what you mentioned your parent of a multi module shouldn't define the dependencies. I assume you mean to define the dependencies within a dependencyManagement section to define the suggested versions for dependencies. All childs can now define dependencies witthout the versions for them.
Apart from that based on the given example from SO the given descriptor will not reproduce the messages your are describing (with maven-assembly-plugin version 2.4). It would be great having an example project which exactly reproduces the behaviour and attach it to this issue.

Furthermore i will close this issue.

If you have supplemental information/example project etc. don't hesitate to reopen the issue.

**Comment by jira-importer** [comment]
**[monkegoist](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=monkegoist)** commented

Personally I don't use dependencyManagement at all, since I find it quite useless. If two or more modules depend on the same library (for example, log4j), I move it to parent pom that all of the children inherit from. If I used dependencyManagement, I'd have to define the log4j version in parent pom, and then explicitly add it to set of dependencies in each of the chlid projects, which is a lot of copy-paste.

But I agree that this issue seems to have been fixed. It is reproducible with 2.3 version of assembly plugin, but no longer reproducible with 2.4 version.
